### PR TITLE
mrc-3406 expose mode control

### PR DESCRIPTION
**Pull request by hillalex**
Expose the stepper control for `mode` models via the particle filter interface.

### CODE REVIEW

File: R/particle_filter.R [hunk 1]
```diff
@@ -210,13 +212,16 @@ particle_filter <- R6::R6Class(
...
+    ##' @param control Tuning control for stepper
     initialize = function(data, model, n_particles, compare,
                           index = NULL, initial = NULL,
                           constant_log_likelihood = NULL,
                           n_threads = 1L, seed = NULL,
                           n_parameters = NULL,
                           gpu_config = NULL,
-                          stochastic_schedule = NULL) {
+                          stochastic_schedule = NULL,
+                          control = NULL) {
```
Comment (hillalex) [inline R/particle_filter.R:224]: this argument name is probably too vague. `stepper_control` better?

Comment (richfitz) [inline R/particle_filter.R:224]: Can we make this `ode_control` perhaps? (not `mode_control` though as we'll eventually merge these)

File: R/particle_filter.R [hunk 2]
```diff
@@ -247,11 +252,18 @@ particle_filter <- R6::R6Class(
...
+          stop(paste("'stochastic_schedule' provided but 'model' does not",
+           "support this"))
+        }
+        if (!is.null(control)) {
+          stop(paste("'control' provided but 'model' does not",
+                     "support this"))
+        }
       } else {
         private$stochastic_schedule <- stochastic_schedule
+        private$control <- control
```
Comment (richfitz) [inline R/particle_filter.R:266]: Add

```
assert_is(control, "mode_control")
```

to validate that the control is ok here, and not when we initialise the mode model later.

(renaming control -> ode_control if you agree with the above)

**Review by richfitz — CHANGES_REQUESTED** [review]
Small comments, nice how small this ends up being.

Can you also bump the version on this (and all mcstate PRs) - it'll fail on merge as it will try and tag the version, the version numbers are used by the downstream science users

File: R/particle_filter.R (hunk 1 above)
Comment (richfitz) [reply R/particle_filter.R]: Sorry totally missed this comment here before :(

**Review by richfitz — APPROVED** [review]

**Pull request merged**